cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,270 @@
+					},
+					SelectorSyncSets: []v1alpha1.SyncStatus{
+						{
+							Name:           "selectorsyncset1",
+							Result:         "Failure",
+							FailureMessage: "Failed to sync selectorsyncset1",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+// Central function to mock the List method and set expectations
+func setupMockClient(mockClient *mockk8s.MockClient, returnErr error) {
+	mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+			switch v := list.(type) {
+			case *hivev1.ClusterDeploymentList:
+				*v = cdList
+			case *v1alpha1.ClusterSyncList:
+				*v = csList
+			}
+			return returnErr
+		}).Times(2) // Expect List to be called twice for ClusterDeployment and ClusterSync
+}
+
+func TestNewCmdClusterSyncFailures(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	streams := genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
+	mockClient := mockk8s.NewMockClient(mockCtrl)
+
+	setupMockClient(mockClient, nil)
+
+	cmd := NewCmdClusterSyncFailures(streams, mockClient)
+	assert.Equal(t, "clustersync-failures [flags]", cmd.Use, "Command use should be 'clustersync-failures [flags]'")
+	assert.Contains(t, cmd.Long, "Helps investigate ClusterSyncs", "Command long description should be set")
+	assert.Contains(t, cmd.Example, "$ osdctl hive csf", "Command example should be set")
+
+	cmd.Flags().Set("limited-support", "true")
+	includeLimitedSupport := cmd.Flags().Lookup("limited-support").Value.String()
+	assert.Equal(t, "true", includeLimitedSupport, "Flag 'limited-support' should be parsed correctly")
+
+	cmd.Flags().Set("hibernating", "true")
+	includeHibernating := cmd.Flags().Lookup("hibernating").Value.String()
+	assert.Equal(t, "true", includeHibernating, "Flag 'hibernating' should be parsed correctly")
+
+	cmd.Flags().Set("syncsets", "false")
+	includeFailingSyncSets := cmd.Flags().Lookup("syncsets").Value.String()
+	assert.Equal(t, "false", includeFailingSyncSets, "Flag 'syncsets' should be parsed correctly")
+
+	cmd.Flags().Set("no-headers", "true")
+	noHeaders := cmd.Flags().Lookup("no-headers").Value.String()
+	assert.Equal(t, "true", noHeaders, "Flag 'no-headers' should be parsed correctly")
+
+	cmd.Flags().Set("output", "yaml")
+	output := cmd.Flags().Lookup("output").Value.String()
+	assert.Equal(t, "yaml", output, "Flag 'output' should be parsed correctly")
+
+	cmd.Flags().Set("cluster-id", "1234")
+	cmd.Flags().Set("limited-support", "true")
+	cmd.Flags().Set("hibernating", "false")
+	cmd.Flags().Set("syncsets", "true")
+	cmd.Flags().Set("output", "json")
+
+	err := cmd.Execute()
+	assert.NoError(t, err, "Command should execute without error")
+
+	err = cmd.Flags().Set("invalid-flag", "true")
+	assert.Error(t, err, "Setting invalid flag should return an error")
+}
+
+func testListFailingClusterSyncs(t *testing.T, returnErr error, expectedResults []failingClusterSync) {
```

Comment:
Can't this be in the body of `TestListFailingClusterSyncs`?

Reply:
Thanks for the suggestion. I opted to keep the function separate to make the code more modular and easier to read and maintain. I felt that keeping the logic in a separate function helps in simplifying the test and making it more manageable.